src/key.js

```diff
@@ -567,44 +569,47 @@ Key.prototype.update = function(key) {
     }
     this.primaryKey = key.primaryKey;
   }
+  // FIXME should we check the expiration date of a revocation signature?
   // revocation signature
   if (!this.revocationSignature && key.revocationSignature && !key.revocationSignature.isExpired() &&
      (key.revocationSignature.verified ||
-      key.revocationSignature.verify(key.primaryKey, {key: key.primaryKey}))) {
+      await key.revocationSignature.verify(key.primaryKey, {key: key.primaryKey}))) {
     this.revocationSignature = key.revocationSignature;
   }
   // direct signatures
   mergeSignatures(key, this, 'directSignatures');
+  // FIXME find a better design pattern here
   // users
-  key.users.forEach(function(srcUser) {
+  await Promise.all(key.users.map(async function(srcUser) {
     var found = false;
-    for (var i = 0; i < that.users.length; i++) {
-      if (srcUser.userId && (srcUser.userId.userid === that.users[i].userId.userid) ||
-          srcUser.userAttribute && (srcUser.userAttribute.equals(that.users[i].userAttribute))) {
-        that.users[i].update(srcUser, that.primaryKey);
+    await Promise.all(that.users.map(async function(dstUser) {
+      if ((srcUser.userId && (srcUser.userId.userid === dstUser.userId.userid)) ||
+          (srcUser.userAttribute && (srcUser.userAttribute.equals(dstUser.userAttribute)))) {
+        await dstUser.update(srcUser, that.primaryKey);
         found = true;
-        break;
+//        break;
```

Comment:
If you are going the Promise.all route here, you should remove this commented out break. Otherwise, you have the option of doing this sequentially rather than with a Promise.all.

Reply:
Will do. I stared at it for 10 minutes to remember why I didn't just remove it, but can't think of anything (and I would have left a FIXME somewhere if there was anything to consider anyway).